src/screens/NativeIntegration/NIInputService.tsx

```diff
@@ -0,0 +1,279 @@
+  const feeInFiatDisplayValue = `${currencySymbol}${feeInFiat.toFixed(5)}`;
+
+  const [value, setValue] = React.useState(blankArr);
+  const [contractRes, setContractRes] = React.useState([]);
+  const [integrationContract, setIntegrationContract]: any = React.useState();
+  const [isSendTransaction, setIsSendTransaction] = React.useState(false);
+  const [amount, setAmount] = React.useState(null);
+
+  React.useEffect(() => {
+    dispatch(fetchGasInfoAction(chain));
+    const integrationCon = etherspotService.getContract(chain, contractData?.abi, contractData?.contract_address);
+    setIntegrationContract(integrationCon);
+  }, [dispatch]);
+
+  const updateTxFee = async () => {
+    if (!value) return;
+    if (contractType === 'payable' ? (amount === null ? true : false) : false) return;
+    const findNull = value.find((res) => res === null || res === '');
+    if (findNull !== undefined) {
+      setContractRes(undefined);
+      return;
+    }
+    await contractInterface();
+  };
+
+  const updateTxFeeDebounced = React.useCallback(debounce(updateTxFee, 100), [value, amount]);
+
+  React.useEffect(() => {
+    updateTxFeeDebounced();
+    return updateTxFeeDebounced.cancel;
+  }, [updateTxFeeDebounced]);
+
+  const contractInterface = async () => {
+    const fnName = actionName ? `encode${actionName[0].toUpperCase()}${actionName.substring(1)}` : ``;
+    const updatedArr = value.map(async (specificVal, i) =>
+      specificVal?.c ? getDecimalValue(specificVal, i) : specificVal,
+    );
+
+    try {
+      setIsSendTransaction(true);
+      const valueWithBigNumber = await Promise.all(updatedArr);
+
+      const list = await approvalList();
+      const approveTxList = list ? await Promise.all(list) : [];
+
+      const contractInterface = await integrationContract[fnName](...valueWithBigNumber);
+
+      setContractRes([
+        ...approveTxList,
+        { ...contractInterface, value: isPayable ? utils.parseUnits(amount.toString(), 18) : 0 },
+      ]);
+
+      const approveListWithValue = approveTxList.map((response) => {
+        return { ...response, value: 0 };
+      });
+
+      dispatch(
+        estimateTransactionsAction(
+          [...approveListWithValue, { ...contractInterface, value: isPayable ? amount?.toString() : 0 }],
+          chain,
+        ),
+      );
+      setIsSendTransaction(false);
+      logBreadcrumb('nativeIntegrationContractResponse', JSON.stringify(contractInterface));
+    } catch (e) {
+      setIsSendTransaction(false);
+      setContractRes(undefined);
+      Toast.show({
+        message: e,
+        emoji: 'warning',
+      });
+      logBreadcrumb('contractInput error!', e);
+    }
+  };
+
+  const approvalList = async () => {
+    const approvalFilter = sequence?.filter((res) => res?.isApprovalNeeded);
+
+    const listOfApproval = approvalFilter
+      ? await approvalFilter.map(async (approvalData, i) => {
```

Comment:
We've got a huge amount of logic here handing off the falsey statement here. Could we encapsulate this in a function?